internal/bminventory/inventory_test.go

```diff
@@ -302,6 +302,149 @@ var _ = Describe("GenerateClusterISO", func() {
 			})
 			verifyApiError(generateReply, http.StatusInternalServerError)
 		})
+
+		It("static ips config - success", func() {
+			staticIPsConfig := []*models.StaticIPConfig{{DNS: "dns1",
```

Comment:
please put common code in a `Context`

Reply:
those is an input parameters for a specific tests, i am not sure it is supposed to be in a general context

Reply:
Not in general, but if you have several tests about same topic you can group then, context can be used multiple times.

Reply:
you want me to move the staticIpsConfig, or the tests themselves? The tests are part of unit test for GenerateClusterISO, seems strange to  open an additional context just for those 2

Reply:
no, you are over thinking it
```go
		Context("status ip", func() {
			staticIPsConfig := []*models.StaticIPConfig{{DNS: "dns1",
				Gateway: "gateway1",
				IP:      "IP1",
				Mac:     "mac1",
				Mask:    "mask1",
			},
				{DNS: "dns2",
					Gateway: "gateway2",
					IP:      "IP2",
					Mac:     "mac2",
					Mask:    "mask2",
				},
			}

			It("static ips config - success", func() {
				clusterId := registerCluster(true).ID
				mockS3Client.EXPECT().GetObjectSizeBytes(gomock.Any(), gomock.Any()).Return(int64(100), nil).Times(1)
				mockS3Client.EXPECT().Upload(gomock.Any(), gomock.Any(), fmt.Sprintf("%s/discovery.ign", clusterId))
				mockS3Client.EXPECT().UploadISO(gomock.Any(), gomock.Any(), fmt.Sprintf("discovery-image-%s", clusterId.String()))
				mockEvents.EXPECT().AddEvent(gomock.Any(), *clusterId, nil, models.EventSeverityInfo, "Generated image (SSH public key is not set)", gomock.Any())
				generateReply := bm.GenerateClusterISO(ctx, installer.GenerateClusterISOParams{
					ClusterID:         *clusterId,
					ImageCreateParams: &models.ImageCreateParams{StaticIpsConfig: staticIPsConfig},
				})
				Expect(generateReply).Should(BeAssignableToTypeOf(installer.NewGenerateClusterISOCreated()))
			})

			It("static ip config  - same static ip config image already exists", func() {
				clusterId := registerCluster(true).ID
				mockS3Client.EXPECT().IsAwsS3().Return(false)
				mockS3Client.EXPECT().GetObjectSizeBytes(gomock.Any(), gomock.Any()).Return(int64(100), nil).Times(1)
				mockS3Client.EXPECT().Upload(gomock.Any(), gomock.Any(), fmt.Sprintf("%s/discovery.ign", clusterId))
				mockS3Client.EXPECT().UploadISO(gomock.Any(), gomock.Any(), fmt.Sprintf("discovery-image-%s", clusterId.String()))
				mockEvents.EXPECT().AddEvent(gomock.Any(), *clusterId, nil, models.EventSeverityInfo, "Generated image (SSH public key is not set)", gomock.Any())
				generateReply := bm.GenerateClusterISO(ctx, installer.GenerateClusterISOParams{
					ClusterID:         *clusterId,
					ImageCreateParams: &models.ImageCreateParams{StaticIpsConfig: staticIPsConfig},
				})
				Expect(generateReply).Should(BeAssignableToTypeOf(installer.NewGenerateClusterISOCreated()))

				updatedTime := time.Now().Add(-11 * time.Second)
				updates := map[string]interface{}{}
				updates["image_created_at"] = updatedTime
				db.Model(&common.Cluster{}).Where("id = ?", clusterId).Updates(updates)

				newStaticIPsConfig := []*models.StaticIPConfig{{DNS: "dns2",
					Gateway: "gateway2",
					IP:      "IP2",
					Mac:     "mac2",
					Mask:    "mask2",
				},
					{DNS: "dns1",
						Gateway: "gateway1",
						IP:      "IP1",
						Mac:     "mac1",
						Mask:    "mask1",
					},
				}

				mockS3Client.EXPECT().UpdateObjectTimestamp(gomock.Any(), gomock.Any()).Return(true, nil).Times(1)
				mockS3Client.EXPECT().IsAwsS3().Return(false)
				mockS3Client.EXPECT().GetObjectSizeBytes(gomock.Any(), gomock.Any()).Return(int64(100), nil).Times(1)
				mockEvents.EXPECT().AddEvent(gomock.Any(), *clusterId, nil, models.EventSeverityInfo, "Re-used existing image rather than generating a new one", gomock.Any())
				generateReply = bm.GenerateClusterISO(ctx, installer.GenerateClusterISOParams{
					ClusterID:         *clusterId,
					ImageCreateParams: &models.ImageCreateParams{StaticIpsConfig: newStaticIPsConfig},
				})

				Expect(generateReply).Should(BeAssignableToTypeOf(installer.NewGenerateClusterISOCreated()))
			})

			It("static ip config  - different static ip config", func() {
				clusterId := registerCluster(true).ID
				mockS3Client.EXPECT().IsAwsS3().Return(false)
				mockS3Client.EXPECT().GetObjectSizeBytes(gomock.Any(), gomock.Any()).Return(int64(100), nil).Times(1)
				mockS3Client.EXPECT().Upload(gomock.Any(), gomock.Any(), fmt.Sprintf("%s/discovery.ign", clusterId))
				mockS3Client.EXPECT().UploadISO(gomock.Any(), gomock.Any(), fmt.Sprintf("discovery-image-%s", clusterId.String()))
				mockEvents.EXPECT().AddEvent(gomock.Any(), *clusterId, nil, models.EventSeverityInfo, "Generated image (SSH public key is not set)", gomock.Any())
				generateReply := bm.GenerateClusterISO(ctx, installer.GenerateClusterISOParams{
					ClusterID:         *clusterId,
					ImageCreateParams: &models.ImageCreateParams{StaticIpsConfig: staticIPsConfig},
				})
				Expect(generateReply).Should(BeAssignableToTypeOf(installer.NewGenerateClusterISOCreated()))

				updatedTime := time.Now().Add(-11 * time.Second)
				updates := map[string]interface{}{}
				updates["image_created_at"] = updatedTime
				db.Model(&common.Cluster{}).Where("id = ?", clusterId).Updates(updates)

				newStaticIPsConfig := []*models.StaticIPConfig{{DNS: "dns11",
					Gateway: "gateway11",
					IP:      "IP11",
					Mac:     "mac11",
					Mask:    "mask11",
				},
					{DNS: "dns22",
						Gateway: "gateway22",
						IP:      "IP22",
						Mac:     "mac22",
						Mask:    "mask22",
					},
				}

				mockS3Client.EXPECT().IsAwsS3().Return(false)
				mockS3Client.EXPECT().GetObjectSizeBytes(gomock.Any(), gomock.Any()).Return(int64(100), nil).Times(1)
				mockS3Client.EXPECT().Upload(gomock.Any(), gomock.Any(), fmt.Sprintf("%s/discovery.ign", clusterId))
				mockS3Client.EXPECT().UploadISO(gomock.Any(), gomock.Any(), fmt.Sprintf("discovery-image-%s", clusterId.String()))
				mockEvents.EXPECT().AddEvent(gomock.Any(), *clusterId, nil, models.EventSeverityInfo, "Generated image (SSH public key is not set)", gomock.Any())

				generateReply = bm.GenerateClusterISO(ctx, installer.GenerateClusterISOParams{
					ClusterID:         *clusterId,
					ImageCreateParams: &models.ImageCreateParams{StaticIpsConfig: newStaticIPsConfig},
				})

				Expect(generateReply).Should(BeAssignableToTypeOf(installer.NewGenerateClusterISOCreated()))
			})
		})
```

Reply:
done